proposals/eligibility_validator.go

```diff
@@ -92,36 +93,48 @@ func (v *Validator) CheckEligibility(ctx context.Context, ballot *types.Ballot)
 		return false, err
 	}
 
-	counter := ballot.EligibilityProof.J
-	if counter >= numEligibleBallots {
-		return false, fmt.Errorf("%w: proof counter (%d) numEligibleBallots (%d), totalWeight (%v)",
-			errIncorrectCounter, counter, numEligibleBallots, totalWeight)
-	}
```

Comment:
also check `len(ballot.EligibilityProofs) == numEligibleBallots`

Reply:
i don't think that there is any guarantee for that

Reply:
for honest parties? why not?

Reply:
i just think this is not the way it works. maybe i misunderstood things, but numEligibleBallots is a boundary for the counter, not the number of ballots  in the layer. e.g. this is a number of ballots in the epoch

Reply:
i stand corrected. apologies.